server/sockets.js

```diff
@@ -567,31 +569,15 @@ if (cluster.isMaster) {
 			room = rooms.get(roomid);
 			if (!room) return;
 
-			/** @type {[string?, string?, string?]} */
-			let messages = [null, null, null];
-			message = data.substr(nlLoc + 1);
+			const split = SPLIT_REGEX.exec(data.substr(nlLoc + 1));
```

Comment:
At this point, `data` is something like

```
>battle-foobar
|start|battle
|split|p1
|switch|p1: Pikachu|350/350
|switch|p1: Pikachu|100/100
```

You want to send to player 1:

```
>battle-foobar
|start|battle
|switch|p1: Pikachu|350/350
```

and to player 2:

```
>battle-foobar
|start|battle
|switch|p1: Pikachu|100/100
```

Reply:
As far as I can tell, your code instead sends to player 1:

```

|switch|p1: Pikachu|350/350
```

and to player 2:

```

|switch|p1: Pikachu|100/100
```

The function I pasted in comments will be useful for your sending the correct message.

Reply:
I think the fundamental misunderstanding I had is that message would be one logical command, not multiple.